optics-core/src/Optics/Operators.hs

```diff
@@ -43,14 +47,33 @@ infixl 8 ^.
 
 infixl 8 ^?
 
+-- | Perform an *UNSAFE* 'head' of an affine fold assuming that it is there.
+--
+-- >>> Left 4 ^?! _Left
+-- 4
+--
+-- >>> "world" ^?! ix 3
+-- 'l'
+--
+-- >>> [] ^?! _head
+-- *** Exception: (^?!): empty affine fold
+-- ...
+--
+-- @since 0.2.1
+(^?!) :: (HasCallStack, Is k An_AffineFold) => s -> Optic' k is s a -> a
+s ^?! o = fromMaybe (error "(^?!): empty affine fold") (s ^? o)
```

Comment:
It calls `error`, so IMHO should be in some `Unsafe` module.

Reply:
Makes sense, I moved it to `Optics.Operators.Unsafe`.